### PR TITLE
added higher-order-function.md in javascript

### LINKED ISSUE #7326: Add Higher Order Function in Function category in JavaScript roadmap
### Roadmap URL

https://roadmap.sh/javascript

### Suggestions

Add Higher Order Function in Function category in JavaScript roadmap.
image

### PR DESCRIPTION
**Pull request by UdaySharmaGitHub**
fix #7326 
- added higher-order-function.md in javascript.
- And raise Pull Request please accept under Hacktoberfest or Hacktoberfest2024

### CODE REVIEW

File: src/data/roadmaps/javascript/content/higher-order-function.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# JavaScript Higher Order Functions
+JavaScript Higher-Order Functions are functions that can accept other functions as arguments, return functions, or both. They enable abstraction and flexibility in code, allowing you to create reusable and modular functions for complex operations, making them essential in functional programming.
```
Comment (arikchakma) [inline src/data/roadmaps/javascript/content/higher-order-function.md:2]:
```suggestion

JavaScript Higher-Order Functions are functions that can accept other functions as arguments, return functions, or both. They enable abstraction and flexibility in code, allowing you to create reusable and modular functions for complex operations, making them essential in functional programming.
```

File: src/data/roadmaps/javascript/content/higher-order-function.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    // Call the callback function
+    callback();
+}
+function callbackFunction() {
+    console.log("Callback function is executed.");
+}
+// Passing the callback function to the higher-order function
+higherOrderFunction(callbackFunction);
+```
+Visit the following resources to learn more:
```
Comment (arikchakma) [inline src/data/roadmaps/javascript/content/higher-order-function.md:17]:
```suggestion

Visit the following resources to learn more:

```

**Comment by UdaySharmaGitHub** [comment]
fix #7347 
- Done All the Suggested Changes
- added higher-order-function.md in javascript.
- And raise Pull Request please accept under Hacktoberfest or Hacktoberfest2024

**Comment by dansholds** [comment]
You've just created a new markdown file here, this won't render anywhere.

Please review the contribution docs thoroughly, especially the Hacktoberfest section.

**Pull request closed without merging**